addons/product/models/product.py

```diff
@@ -747,6 +747,52 @@ def toggle_active(self):
         (tmpl_to_deactivate + tmpl_to_activate).toggle_active()
         return result
 
+    def _get_real_price_currency(self, rule_id, qty=1, uom=None, partner_id=None, company=None, date=None):
```

Comment:
Makes sense since the price&discount computation logic should ideally be fully defined in `product`.
(I really don't like this old method whose naming and code is quite confusing, but it'll be cleaned soon in master anyway, no need to clean it here).

Maybe add some tests to make sure all the pricelist changes in the next versions do not break this fix :).

Reply:
> Makes sense since the price&discount computation logic should ideally be fully defined in product.
(I really don't like this old method whose naming and code is quite confusing, but it'll be cleaned soon in master anyway, no need to clean it here).

Definitely agree here, but makes sense to keep a minimal diff in stable :+1: 

> Maybe add some tests to make sure all the pricelist changes in the next versions do not break this fix :).

To me it makes much more sense to have some `_get_real_price_currency` unit tests (maybe there are some already, didn't find any though?). I don't really see the fix breaking otherwise, since we won't be moving back this piece of logic to `sale`.

Reply:
There should be some tests in sale to verify the discount is correctly computed on sales orders, but there isn't any test focused on the method itself.